realm/realm-library/src/main/java/io/realm/internal/SharedRealm.java

```diff
@@ -305,7 +306,37 @@ public Table getTable(String name) {
      * @return a created {@link Table} object.
      */
     public Table createTable(String name) {
-        return new Table(this, nativeCreateTable(nativePtr, name));
+        return new Table(this, nativeCreateTable(nativePtr, name, false));
+    }
+
+    /**
+     * Creates a primary key table with then given name. Native assertion will happen if the table with the same name
+     * exists. This function is different from {@link #createTable(String)} which will call {@code create_table()} from
+     * sync to do the creation. This will always call the core's {@code add_table()} to avoid creating the stable id
+     * column for pk table.
+     *
+     * @param name the name of table.
+     * @return a created {@link Table} object.
+     */
+    public Table createPkTable(String name) {
+        return new Table(this, nativeCreateTable(nativePtr, name, true));
```

Comment:
the `name` param should be removed, this is only used for creating PK table which the name is constantly `PRIMARY_KEY_TABLE_NAME`. Yes, Object Store is using the same logic. https://github.com/realm/realm-object-store/blob/2aa939f1c0a2f849f67f76ec6a736e7dae36c1ea/src/object_store.cpp#L57
But, we really really really should call OS to do that. After OS schema migration, i will change this.